### PR TITLE
refactor(animations): add unit test for leaving animated child

### PR DESCRIPTION
**Pull request by dario-piotrowicz**
add a new unit test to make sure that leaving elements queried via
a parent queried via animateChild are correctly removed

note: this tests the fix introduced in PR #44357

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

- @jessicajaniuk :slightly_smiling_face::+1:
-  I've run this on an outdated branch and it does fail on code older than #44357 and succeeds on newer code

### CODE REVIEW

File: packages/core/test/animation/animation_query_integration_spec.ts [hunk 1]
```diff
@@ -3348,6 +3348,51 @@ describe('animation query tests', function() {
...
+          `,
+           animations: [
+             trigger(
+                 'grandParentAnimation',
+                 [transition('true <=> false', query('@parentAnimation', animateChild()))]),
+             trigger(
+                 'parentAnimation',
+                 [transition(
+                     'true => false', query(':leave.child', animate('.2s', style({opacity: 0}))))])
+           ]
```
Comment (dario-piotrowicz) [inline packages/core/test/animation/animation_query_integration_spec.ts:3371]: the formatting here may look a bit strange.... that's how the auto-formatter does it on commit :sweat:

File: packages/core/test/animation/animation_query_integration_spec.ts [hunk 2]
```diff
@@ -3348,6 +3348,51 @@ describe('animation query tests', function() {
...
+         let children = fixture.debugElement.nativeElement.querySelectorAll('.child');
+         expect(children.length).toEqual(1);
+
+         cmp.childPresent = false;
+         fixture.detectChanges();
+         flushMicrotasks();
+
+         engine.players.forEach(player => player.finish());
+
+         children = fixture.debugElement.nativeElement.querySelectorAll('.child');
```
Comment (dario-piotrowicz) [inline packages/core/test/animation/animation_query_integration_spec.ts:3393]: I could do a simple `querySelector` and check for `undefined`, but I find it more clear with `querySelectorAll` and checking that the length is 0, but if `querySelector` would be preferred I don't mind changing to that :slightly_smiling_face:

File: packages/core/test/animation/animation_query_integration_spec.ts (hunk 1 above)
Comment (jessicajaniuk) [reply packages/core/test/animation/animation_query_integration_spec.ts:3371]: Yeah, thankfully it looks fine ))))))])])

File: packages/core/test/animation/animation_query_integration_spec.ts (hunk 2 above)
Comment (jessicajaniuk) [reply packages/core/test/animation/animation_query_integration_spec.ts:3393]: They're essentially doing the same thing. So it's fine to go either way.

**Review by jessicajaniuk — APPROVED** [review]
LGTM :cookie:

Thanks, Dario!

reviewed-for: fw-animations, fw-core

**Comment by dylhunn** [comment]
This PR was merged into the repository by commit f2b858d24f9af44e432850d22637aafa178ca382.

**Pull request closed without merging**